### ISSUE TITLE
Items in the Menu Bar are invisible on Ubuntu 18.10 when atom window is focused.

### ISSUE THREAD

**Issue by bagratte** [atom/atom#18535]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Items in the Menu Bar are invisible on Ubuntu 18.10 when atom window is focused.

### Steps to Reproduce

1. Launch Atom.

**Expected behavior:** Menu Bar items (File, Edit, etc) visible.

**Actual behavior:** Menu Bar items invisible.

**Reproduces how often:** Always.

### Versions
Atom    : 1.33.0
Electron: 2.0.11
Chrome  : 61.0.3163.100
Node    : 8.9.3

apm  2.1.2
npm  6.2.0
node 8.9.3 x64
atom 1.33.0
python 2.7.15+
git 2.19.1

Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic

Linux buddy 4.18.0-11-generic #12-Ubuntu SMP Tue Oct 23 19:22:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

Ubuntu is apt-upgraded as of today.

### Additional Information
![screenshot from 2018-12-03 13-09-52](https://user-images.githubusercontent.com/11438666/49364391-af1ef300-f6fd-11e8-8557-d21a98efc6ca.png)
![screenshot from 2018-12-03 13-11-23](https://user-images.githubusercontent.com/11438666/49364402-b3e3a700-f6fd-11e8-8ba3-4df61a1e5dd3.png)

**Comment by bliepp** [comment]
For me, too. Updated today. (Ubuntu 18.10, too)

**Comment by Seb182** [comment]
For me, too. Updated today. (Ubuntu 18.10, too, snap)

**Comment by njee** [comment]
I had the same problem! After tweaking ubuntu to default theme for applications (Adwaita) the menu text reappeared! Hope this helps?

**Comment by upatricck** [comment]
Same issue, after new atom installation.

**Comment by rsese** [comment]
Looks like this is an Electron issue:

https://github.com/electron/electron/issues/15194 and https://github.com/electron/electron/issues/15194#issuecomment-442445138

**Comment by jeremie2** [comment]
I have tried the other themes available in the Gnome Tweak Tool and they work perfectly, the menu bar is visible.
So this problems seems to be related to the default theme "Yaru". Also the dark variant "Yaru-dark" works as expected.

**Comment by williamcanin** [comment]
Same problem here. I hope they solve this, I like the clear theme in Ubuntu 18.10, and only the dark theme does not stay that way. If someone knows how to get around without changing the theme, thank you.

**Comment by tekvsakdan** [comment]
Ubuntu 18.10. Menu bar is invisible if Adapta theme is enabled. It works fine with the default theme Yaru.

**Comment by zypro** [comment]
> Ubuntu 18.10. Menu bar is invisible if Adapta theme is enabled. It works fine with the default theme Yaru.

Seems to be an issue "caused" by Ubuntu themes?

**Comment by rzelnik** [comment]
I don't know if the issue is on the Ubuntu side, or on the Atom side. There is a basic web design principle which says that if your rewrite a text colour, you should also rewrite a background colour. And vice versa, of course.

**Comment by Trilochan20** [comment]
Same thing happened for me. Upgrade today to 18.10 from 18.04. Uninstalled the snap version and downloaded and installed the official .deb package. Still didn't work. 
After changing the theme from Yaru to Yaru-Dark. The menu bar is back.
![atom_menu](https://user-images.githubusercontent.com/29045013/49939805-d64aa100-ff03-11e8-8c83-de8e941c1f66.png)
![theme](https://user-images.githubusercontent.com/29045013/49939806-d64aa100-ff03-11e8-9ed8-aeb92dc8f448.png)

**Comment by cmdcolin** [comment]
Just to fix the link @jeremie2 pointed to I believe it is https://github.com/electron/electron/issues/15878

**Comment by ataraxus** [comment]
any idea when this lands in atom? Or where to track it? just updated to atom 1.34 but it's still broken

**Comment by rsese** [comment]
Based on https://github.com/electron/electron/pull/15878, I *believe* the fix is in Electron 3 so you can subscribe to https://github.com/atom/atom/pull/18603 @ataraxus.

**Comment by joshcangit** [comment]
It's an issue with the way the application interacts with the system theme. Some dark themes on Linux will cause Electron based applications to color the menu bar black on dark grey or worse, black on black.
![screenshot atom](https://user-images.githubusercontent.com/33022160/50558365-9b0ee700-0ce5-11e9-8afc-e6443e6a6127.png)
Should check out [Typora's Issue #1923](https://github.com/typora/typora-issues/issues/1923). There was also this problem until it was fixed.

**Comment by joshcangit** [comment]
About page shows `Atom 1.34.0` is still using `Electron 2.0.16`.
![atom about](https://user-images.githubusercontent.com/33022160/51105934-4e95d200-17e2-11e9-96a2-d68efe708372.png)
It'll be really disappointing if Atom does not update it's dependencies for following updates.

**Comment by Aerijo** [comment]
@joshcangit The upgrade to Electron v3 is in progress [here](https://github.com/atom/atom/pull/18603)

**Comment by joshcangit** [comment]
Great! Looking forward to it.

**Comment by ffais** [comment]
i have same problem in atom 1.34.0 on ubuntu 18.10.

**Comment by Cobinja** [comment]
This should be fixed whenever Atom hits Electron 3.0.14, 3.1.1 or 4.0.1 (or higher)

**Comment by trungpt25** [comment]
[13381#issuecomment-415680837](https://github.com/electron/electron/issues/13381#issuecomment-415680837)
I think this would be temporary fix for now.
Btw I'm using Ubuntu 18.04 with Communitytheme installed through snap and had same issue
![screenshot from 2019-01-23 17-48-56](https://user-images.githubusercontent.com/37434290/51601573-59193f80-1f37-11e9-9827-503f5ee099e4.png)
![screenshot from 2019-01-23 17-46-27](https://user-images.githubusercontent.com/37434290/51601572-59193f80-1f37-11e9-94f5-281d4644c2cd.png)
![screenshot from 2019-01-23 17-53-51](https://user-images.githubusercontent.com/37434290/51601779-de9cef80-1f37-11e9-8ee8-ee7454f0f82f.png)

**Comment by daviwil** [comment]
This bug has been fixed in the latest releases of Electron 3 and 4 but apparently the fix wasn't backported to Electron 2.  The issue will be resolved with our Electron 3 upgrade in next month's Atom 1.36.0-beta0 release (#18603).  Here's a screenshot:

![ubuntu_menu_foreground](https://user-images.githubusercontent.com/79405/51951540-dcc8a580-23e9-11e9-819c-c422bd0e4859.png)

**Comment by Cobinja** [comment]
@daviwil I did the backports of the original electron PR to 3.0, 3.1 and 4.0.
If necessary, I could provide backports to electron 2.0 as well, but it would have to go through the respective electron processes first. Not sure though if it could be done in time for Atom 1.35

**Comment by TriMoon** [comment]
Perhaps the solution provided below will work for those with issues on atom and other apps:
https://askubuntu.com/questions/8336/how-can-one-make-firefox-ignore-my-gtk-theme-entirely#answer-8346

**Comment by rat** [comment]
For me, too. Updated today. (Ubuntu 18.10, too)

**Comment by rsese** [comment]
The fix will be in the next Atom stable and beta releases.

**Comment by manudroid19** [comment]
#18908 looks like we have to wait more for electron 3...

**Comment by rsese** [comment]
Thanks for referencing that @manudroid19 :+1:  Here's the new Electron 3 pull request if folks want to subscribe: https://github.com/atom/atom/pull/18916.

**Comment by colinl** [comment]
Why is this marked as closed when it is still a live issue?  The version of atom picked up by Ubuntu from
`https://packagecloud.io/AtomEditor/atom/any any/main amd64`
 is still 1.35.1.  Or am I getting it from the wrong place?

**Comment by manudroid19** [comment]
Developers have nothing to do with this issue, we just have to wait for Electron 3 to be implemented as noted in previous messages

**Comment by colinl** [comment]
As I understand it the bug is still present in the latest version of Atom (even though it is not in the Atom code itself).  The fact that it is closed means users who come across the problem have to know to look in Closed bugs in order to find it, even though they are using the latest code.  However if that is the way the bug system is used then that is the way it is used.

**Comment by 50Wliu** [comment]
Thanks for bringing that up, @colinl. I'll reopen the issues that #18603 was supposed to fix as that has been reverted, and add them to the list of issues that will be fixed in #18916.

**Comment by StephanX** [comment]
Can confirm, bug still persists in v1.36.0-beta1 + ubuntu 18.10.  Switching to Yaru-Dark does the trick for now.  Thanks!

**Comment by kilazer** [comment]
I have the same problem, the menu is not visible. Version 1.32.2 works well and the menu is visible.
![Снимок экрана от 2019-03-18 13-32-37](https://user-images.githubusercontent.com/23651271/54525798-10559400-4986-11e9-8012-4d099c668b7f.png)

**Comment by datenstau** [comment]
So when is the fix from Electron 3 and 4 coming to Atom?

**Comment by chankruze** [comment]
Same issue.
Wanted to try atom cause in my old windows 8.1 pc it was lagging. But disappointed now.

![Screenshot from 2019-03-26 06-59-25](https://user-images.githubusercontent.com/29806845/54965030-ce83a980-4f94-11e9-874e-c7f3030265b4.png)


```
chankruze@chankruze-TUF:~/Downloads$ atom --version
Atom    : 1.35.1
Electron: 2.0.18
Chrome  : 61.0.3163.100
Node    : 8.9.3
```

**Comment by timburgess** [comment]
Same issue.
I've been using 18.10 since Oct 2018 and the 'no menus' issue appeared after upgrading Atom a few weeks back.
I've installed the beta but the issue is still present. I don't have any theme changes/tweaks so default Ubuntu shows no menus.

![Screenshot from 2019-03-30 11-50-39](https://user-images.githubusercontent.com/28915/55269688-19c5f100-52e2-11e9-90a3-37793d29d5ae.png)

**Comment by qarlosalberto** [comment]
I have the same problem in Linux Mint 19.1 Tessa

```
cruiz@DASPC071:~/Descargas/qfsm$ apm -v
apm  2.1.3
npm  6.2.0
node 8.9.3 x64
atom 1.35.1
python 2.7.15rc1
git 2.17.1
```

**Comment by Pascalrender** [comment]
![49939806-d64aa100-ff03-11e8-9ed8-aeb92dc8f448](https://user-images.githubusercontent.com/27871013/56456465-7db46600-6332-11e9-9fda-2bf1ff1f772f.png)

works in Ubuntu 19.04

**Comment by mlamlu** [comment]
Same problem with ubuntu 19.04 (update from 18.10 with the problem too) If my themes  are Yaru (not dark with dark no problem)
last version installed with snapd

**Comment by nerburish** [comment]
Same problem with ubuntu 18.10 and ubuntu 19.04

**Comment by johnwatterlond** [comment]
same problem ubuntu 19.04. atom version 1.36.1.  also if it helps anyone, i noticed that the menu bar becomes visible if the atom window is behind another window.  when atom window is brought to the front the menu bar becomes invisible again.

**Comment by dwivedisshyam** [comment]
Just change the theme of Ubuntu from Yaru to any other. It will help.

**Comment by RVidalki** [comment]
**In Ubuntu, try this to make Atom Menu visible **

1- Install **gnome-tweak-tool** 
    by typing: sudo apt install gnome-tweak-tool.
2- Install or download additional themes.
3- Start **gnome-tweak-tool.**
4- Select a different theme than **Yaru**    
    Appearance > Themes > Choose theme Applications or Shell from the drop down menu.

**Comment by kilazer** [comment]
> Just change the theme of Ubuntu from Yaru to any other. It will help.

It seems a bit strange to change the topic because of the bug. Why developers are not trying to fix the problem, this has been known for a long time.

**Comment by mhterres** [comment]
It is a workaround, but it would be nice to have a proper fix.

Regards,
Marcelo H. Terres <mhterres@gmail.com>
https://www.mundoopensource.com.br
https://twitter.com/mhterres
https://linkedin.com/in/marceloterres

**Comment by datenstau** [comment]
So when does atom migrate to a new version of electron? This bug is already fixed there in recent versions.

**Comment by colinl** [comment]
As I understand it the upgrade to Electron 3 causes some problems, the upgrade is being addressed in the issue linked earlier
https://github.com/atom/atom/pull/18916

**Comment by rat** [comment]
> For me, too. Updated today. (Ubuntu 18.10, too)


When I opened this I still dreamed of leaving the sublime text, today I see that it is not possible

**Comment by rsese** [comment]
As mentioned earlier in this thread, this should be fixed in the Electron 3 upgrade.  There's no specific timeline for when that will happen, but you can subscribe to https://github.com/atom/atom/issues/18535 if you want to follow the progress.

**Comment by rusikf** [comment]
Hi guys, Does it still possible to fix this without changing theme to Yark dark ? ( for example may be I can use dev build )

**Comment by davehowson** [comment]
Is there any fix to this yet?
Also, how do I update the Electron version? Mine says its 2.0.18 but the screenshot by David says 3.0.14

> This bug has been fixed in the latest releases of Electron 3 and 4 but apparently the fix wasn't backported to Electron 2. The issue will be resolved with our Electron 3 upgrade in next month's Atom 1.36.0-beta0 release (#18603). Here's a screenshot:
> 
> ![ubuntu_menu_foreground](https://user-images.githubusercontent.com/79405/51951540-dcc8a580-23e9-11e9-819c-c422bd0e4859.png)

**Comment by codingbybl** [comment]
In Core Settings, just uncheck "Autohide Menubar".

![GH-submit-atom-menubar-1905](https://user-images.githubusercontent.com/46179830/58219358-6107a700-7cbf-11e9-9e97-74e83c57dfb0.png)

**Comment by colinl** [comment]
@codingbybl that is not the cause of this problem.  It is not that the menu is being hidden, the menu is there but the text is invisible.  Clicking in the area where the menu bar should be seen does work correctly.

**Comment by ludovicdeluna** [comment]
I use Ubuntu 19.04 and have the same issue. I think this is a bug with the default theme of Ubuntu and not directly related to Atom.

The default theme is Yaru and other apps have the same issue than Atom with this theme. I've noticed also a bad choice of font color and bg color for the focused item in the contextual menu outside Atom (really not easy to read). I hope Canonical (or other) will fix this them soon.

**You can fix yourself this issue by doing one of this action**

- Revert back your theme to Ambiance (the default theme for previous Ubuntu version).
- Use Yaru-dark if you want the dark mode.

The new version of Ubuntu don't let you the possibility to easily change your theme. You have to install Gnome Tweak Tool before (what a shame).

```
sudo apt-get install gnome-tweak-tool
```

Launch the app and go to `appearance`: select the theme you wish to use for "Applications" (Ambiance or Yaru-dark). When done, your Atom menu bring back to you.

**Comment by timburgess** [comment]
@ludovicdeluna instructtions are on the right path. Once you've installed `gnome-tweak-tool` it's available as `gnome-tweaks`

I did a fresh install of 19.04 and unfortunately `Ambiance` is not available as a theme. A search of `packages.ubuntu.com` on `ambiance` or `theme` shows no relevant packages. `yaru-dark` makes the Atom menu reappear but unfortunately it makes chromium tabs unreadable IMHO.

**Comment by ludovicdeluna** [comment]
@timburgess : Ok, perhaps you can get it with : 

```
sudo apt-get install light-themes
```

A `apt-cache show light-themes` return this for me:

```
Package: light-themes
Architecture: all
Version: 19.04-0ubuntu1
Multi-Arch: foreign
Priority: optional
Section: universe/gnome
Source: ubuntu-themes
Origin: Ubuntu
Maintainer: Ubuntu Artwork Team <ubuntu-art@lists.ubuntu.com>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 17247
Depends: gtk2-engines-murrine (>= 0.90.3+git20100810), humanity-icon-theme, ubuntu-mono
Filename: pool/universe/u/ubuntu-themes/light-themes_19.04-0ubuntu1_all.deb
Size: 458280
MD5sum: 7ed04a70c6d10efd2c29561d88b26a3f
SHA1: 31bcdf1744b101b74e349293d5c4aa1d18c2c770
SHA256: 627d26bbe1d3060edcff2e8e5b941c978b20754d90d14e293b897e6f7899885f
Homepage: https://launchpad.net/ubuntu-themes
Description-en: Light Themes (Ambiance and Radiance)
 Includes matching Ambiance and Radiance themes
 .
 * Ambiance is a light-on-dark theme
 * Radiance is a dark-on-light theme
 .
 Introduced as the default themes in Ubuntu 10.04 LTS.
Description-md5: 66093fe5a19671849351e8d6fad66909
Task: ubuntukylin-desktop
```

I hope this will help.

**Comment by iuuuuan** [comment]
I am using Yaru theme on ubuntu 19.04. Workaround for this issue is to modify line Exec in file /usr/share/applications/atom.desktop and add env GTK_THEME=Yaru-dark before /usr/bin/atom %F, for example Exec=env GTK_THEME=Yaru-dark /usr/bin/atom %F

My file /usr/share/applications/atom.desktop looks like this

```
[Desktop Entry]
Name=Atom
Comment=A hackable text editor for the 21st Century.
GenericName=Text Editor
Exec=env GTK_THEME=Yaru-dark /usr/bin/atom %F
Icon=atom
Type=Application
StartupNotify=true
Categories=GNOME;GTK;Utility;TextEditor;Development;
MimeType=text/plain;
StartupWMClass=atom
```

Other GTK_THEME can be used as well. Found this on https://discuss.atom.io/t/hardly-visible-menu-on-linux/61816/8

**Comment by psycholein** [comment]
or use an alias for command line: `alias atom='GTK_THEME=Yaru-dark atom'`

**Comment by TriMoon** [comment]
> or use an alias for command line: `alias atom='GTK_THEME=Yaru-dark atom'`

This is IMHO the better work-around because changes to files installed from packages will be overwritten with new versions when they arrive...

**Comment by luixal** [comment]
This bug persists 7 months later... WTF?!

**Comment by rsese** [comment]
Electron 3 is in 1.39 (currently in beta) and the menu bar is ok on Ubuntu 18.10.  I'm assuming the same for 19.04, can anyone running 19.04 confirm with 1.39?

**Comment by mfonville** [comment]
I can confirm that with 19.04 it is works (with Yaru theme)

**Comment by mhterres** [comment]
Me too.

1.39 beta works fine on Ubuntu 19.04


Marcelo H. Terres <mhterres@gmail.com>
https://www.mundoopensource.com.br
https://twitter.com/mhterres
https://linkedin.com/in/marceloterres

**Comment by wesinator** [comment]
Still seeing a similar issue in atom 1.38.2 ([snap](https://snapcraft.io/atom)) with Arc-Darker theme
 - Ubuntu 18.04.2
 - GNOME 3.28.2
 - Arc-Darker theme (20180114-1ubuntu1), set via tweak tool

The menu bar text is always barely visible:
![image](https://user-images.githubusercontent.com/5124946/61177799-43620a00-a5ab-11e9-8654-54bbd4da6a8c.png)

possibly https://github.com/electron/electron/issues/15823 ?

**Comment by Aerijo** [comment]
@wesinator 

1. Please read the above comments. The fix is currently in _beta_, and will likely land in stable on next release.

2. The Snap install is neither official nor supported. So any issues you encounter with it will not be considered until you reproduce them on an official build. If you want a package manager to handle updates, look at the [installing section](https://flight-manual.atom.io/getting-started/sections/installing-atom/#debian-and-ubuntu-debapt) of the flight manual.

**Comment by AllanC56** [comment]
Finally fixed on stable channel with Atom 1.39, yay
